testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPUserLoginTest.java

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.keycloak.testsuite.federation.ldap;
+
+import org.jboss.arquillian.graphene.page.Page;
+import org.junit.FixMethodOrder;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+import org.junit.runners.MethodSorters;
+import org.keycloak.events.EventType;
+import org.keycloak.models.LDAPConstants;
+import org.keycloak.models.RealmModel;
+import org.keycloak.OAuth2Constants;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.storage.ldap.idm.model.LDAPObject;
+import org.keycloak.testsuite.arquillian.annotation.EnableVault;
+import org.keycloak.testsuite.Assert;
+import org.keycloak.testsuite.AssertEvents;
+import org.keycloak.testsuite.pages.AppPage;
+import org.keycloak.testsuite.pages.LoginPage;
+import org.keycloak.testsuite.util.LDAPRule;
+import org.keycloak.testsuite.util.LDAPRule.LDAPConnectionParameters;
+import org.keycloak.testsuite.util.LDAPTestUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test user logins utilizing various LDAP authentication methods and different LDAP connection encryption mechanisms.
+ *
+ * @author <a href="mailto:jlieskov@redhat.com">Jan Lieskovsky</a>
+ */
+@EnableVault
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class LDAPUserLoginTest extends AbstractLDAPTest {
+
+    @Rule
+    // Start an embedded LDAP server with configuration derived from test annotations before each test
+    public LDAPRule ldapRule = new LDAPRule();
```

Comment:
I think this test will work just with embedded ApacheDS server? It seems for other LDAP servers, test will need to be ignored. See for example LDAPLegacyImportTest for the inspiration.

Reply:
Fixed in 2nd commit (intentionally kept it separate, so the changes done are better visible) Will squash the commits later (since it's likely there will be conflicts with the Tero's one, I might need to up2date this one anyway). Thanks for the example test hint, utilizing that assumeTrue() condition!